src/toeplitzsolvers.jl

```diff
@@ -1,28 +1,54 @@
-# Symmetric Toeplitz solver
-function durbin!(r::AbstractVector{T}, y::AbstractVector{T}) where T<:BlasReal
+"""
+    durbin!(r::AbstractVector, y::AbstractVector, p::AbstractVector) -> Vector
+
+Solve Yule-Walker equations using Durbin algorithm.
+
+Solution of NxN system is obtained iteratively, by solving 1x1, 2x2, ...
+in succession. For use in computing partial autocorrelation matrix,
+return the last elements of the successive solutions in vector p[].
+
+The section 4.7 of Golub,VanLoan "Matrix Computations," 4th ed.,
+discusses the algorithm in detail.
+
+# Arguments
+- `r::AbstractVector`: first column of a herimitian positive definite
+    Toeplitz matrix, excluding the diagonal element (equal to one).
+- `y::AbstractVector`: work vector for solution, should have the same length
+    as r[]
+- `p::AbstractVector`: last elements of the successive solutions, should
+     have the same length  as r[]
+
+# Returns
+- `y::AbstractVector`: same as the second argument
```

Comment:
The convention we use is rather to say "return a vector..." in the first sentence.